### ISSUE TITLE
MissingJavadocMethod flagging private methods even target scope is public methods only

### ISSUE THREAD

**Issue by ChrisTrenkamp** [checkstyle/checkstyle#8266]
https://checkstyle.sourceforge.io/config_javadoc.html#MissingJavadocMethod

```
$ javac CheckstylePrivateMethodTest.java
$ 
```
```
$ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name = "Checker">
    <module name="TreeWalker">
        <module name="MissingJavadocMethod">
            <property name="scope" value="public"/>
            <property name="minLineCount" value="2"/>
            <property name="tokens" value="METHOD_DEF"/>
        </module>
    </module>
</module>
```

```
$ cat CheckstylePrivateMethodTest.java
public interface CheckstylePrivateMethodTest {
  private static void whyIsMissingJavadocMethodBeingFlagged() {
    System.out.println();
    System.out.println();
    System.out.println();
    System.out.println();
  }
}
```

```
$ java -jar checkstyle-8.32-all.jar -c config.xml CheckstylePrivateMethodTest.java
Starting audit...
[ERROR] CheckstylePrivateMethodTest.java:2:3: Missing a Javadoc comment. [MissingJavadocMethod]
Audit done.
Checkstyle ends with 1 errors.

```
---------------
Expected - no violation.

When an interface has a private static method that goes beyond the line limit, it gets flagged as needing a javadoc comment, even though private methods are supposed to be exempt.

--------------

**Comment by strkkk** [comment]
@ChrisTrenkamp Please correct your config example to only one failing check.
Issue is in line 
https://github.com/checkstyle/checkstyle/blob/6275303e6b5fd4d5e9a6504b390efd0e4aee564d/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/MissingJavadocMethodCheck.java#L453-L455
interfaced can have private methods since Java 9
So it looks like a bug

**Comment by ChrisTrenkamp** [comment]
> 
> 
> @ChrisTrenkamp Please correct your config example to only one failing check.
> Issue is in line

Done.

**Comment by romani** [comment]
fix is merged.
